File: android/app/src/main/kotlin/com/sadat/jamaattime/familysafety/vpn/BlockCategory.kt
package com.sadat.jamaattime.familysafety.vpn

enum class BlockCategory(val id: Int, val assetPath: String) {
    ADULT(1, "flutter_assets/assets/family_safety/blocklists/adult_2026_05_seed.txt"),
    GAMBLING(2, "flutter_assets/assets/family_safety/blocklists/gambling_2026_05_seed.txt"),
    PROXY_BYPASS(3, "flutter_assets/assets/family_safety/blocklists/proxy_bypass_2026_05_seed.txt");

    companion object {
        fun fromId(id: Int): BlockCategory? = values().firstOrNull { it.id == id }
    }
}

File: lib/features/family_safety/domain/block_category.dart
enum BlockCategory {
  adult(1),
  gambling(2),
  proxyBypass(3),
  focusGuardShortVideo(4);

  const BlockCategory(this.id);

  final int id;

  bool get isWebsiteProtectionCategory {
    return switch (this) {
      BlockCategory.adult ||
      BlockCategory.gambling ||
      BlockCategory.proxyBypass => true,
      BlockCategory.focusGuardShortVideo => false,
    };
  }
}

const Set<BlockCategory> websiteProtectionBlockCategories = <BlockCategory>{
  BlockCategory.adult,
  BlockCategory.gambling,
  BlockCategory.proxyBypass,
};
